plugins/catalog-backend/src/database/operations/util/deleteOrphanedEntities.ts

```diff
@@ -72,26 +75,17 @@ export async function deleteOrphanedEntities(options: {
     total += orphanIds.length;
 
     // Delete the orphans themselves
-    await tx
+    await knex
       .table<DbRefreshStateRow>('refresh_state')
       .delete()
       .whereIn('entity_id', orphanIds);
 
-    // Mark all of things that the orphans had relations to for processing and
-    // stitching
-    await tx
-      .table<DbFinalEntitiesRow>('final_entities')
-      .update({
-        hash: 'orphan-relation-deleted',
-      })
-      .whereIn('entity_id', orphanRelationIds);
-    await tx
-      .table<DbRefreshStateRow>('refresh_state')
-      .update({
-        result_hash: 'orphan-relation-deleted',
-        next_update_at: tx.fn.now(),
-      })
-      .whereIn('entity_id', orphanRelationIds);
+    // Mark all of the things that the orphans had relations to for stitching
+    await markForStitching({
```

Comment:
maybe I'm misunderstanding, but wouldn't this introduce the deferred stitching?

Reply:
No, even in the old system there was a way to mark something for stitching _kind of_, which is what this retains. No functional changes as such. What it does is to mark the entity for both re-processing and re-stitching.

In the later PR, it'll have an if statement in that function doing it the old way or the more efficient new way, depending on settings